src/main/java/io/github/nihaljain/Util.java

```diff
@@ -40,16 +41,12 @@ public static String getLogDirPath(String[] args) throws Exception {
     logger.info("useCanonicalPath flag: " + useCanonicalPath);
 
     // Get the final log path for the log directory
-    String logDirPath = useCanonicalPath ? Paths.get(logDir).toFile().getCanonicalPath() : logDir;
+    Path logDirPath = useCanonicalPath ? Paths.get(logDir).toRealPath() : Paths.get(logDir);
```

Comment:
```suggestion
    Path logDirPath = useCanonicalPath ? Path.of(logDir).toRealPath() : Path.of(logDir).toAbsolutePath();
```
Use `Path.of` not `Paths.get` (check the javadoc on Paths.get, it's just uses `Path.of` anyway)
Also, use the Absolute Path in the non-canonical condition.